src/ml_board/frontend/Dashboard/src/redux/charts/chartsSlice.ts

```diff
@@ -72,16 +72,21 @@ export const { upsertCharts, resetChartState } = actions;
 // TODO: memoize these selectors
 export const selectChartLabelsById = (state: RootState, chart_id: string) => state.charts.entities[chart_id]?.x_axis ?? [];
 export const selectExperimentsPerChartById = (state: RootState, chart_id: string) => state.charts.entities[chart_id]?.experiments.entities ?? {};
-export const selectChartsByExperimentId = (state: RootState, exp_id: string) =>{
-    const allExp: {[key: string]: any} = {};
-    for (const chart_id in state.charts.entities) {
-        const experiments = state.charts.entities[chart_id]?.experiments.entities ?? {};
-        if(experiments[exp_id]) {
-            allExp[chart_id] = experiments[exp_id];
+// memoized to avoid unnecessary rerendering
+export const selectChartsByExperimentId = createSelector(
```

Comment:
Oh I see, memoization will do a comparison of the previous props with the new props, and if they're the same, it skips rendering else it will re-render the charts components. Thus in case of completed experiments, the graph component will not be rendered again repeatedly... right? Nice observation of this bug! @le1nux  looks good. We can merge it